### PR TITLE
merkaartor: build with cmake

### PR DESCRIPTION
**Pull request by wegank**
## Description of changes

https://github.com/openstreetmap/merkaartor/blob/0.19.0/Merkaartor.pro

> qmake support is deprecated, please use cmake build system whenever possible. qmake support will be removed in 0.20.0 release.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/merkaartor/default.nix [hunk 1]
```diff
@@ -34,29 +35,26 @@ stdenv.mkDerivation rec {
...
+    (lib.cmakeBool "ZBAR" withZbar)
+  ];
 
   postInstall = lib.optionalString stdenv.isDarwin ''
-    mkdir -p $out/Applications
-    mv binaries/bin/merkaartor.app $out/Applications
-    mv binaries/bin/plugins $out/Applications/merkaartor.app/Contents
+    mkdir -p $out/{Applications,bin}
+    mv $out/merkaartor.app $out/Applications
+    makeWrapper $out/{Applications/merkaartor.app/Contents/MacOS,bin}/merkaartor
```
Comment (sikmir) [inline pkgs/applications/misc/merkaartor/default.nix:57]: Consider adding `meta.mainProgram = "merkaartor";`

Comment (wegank) [reply pkgs/applications/misc/merkaartor/default.nix:57]: Done.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 259434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>merkaartor</li>
  </ul>
</details>

```
error: builder for '/nix/store/b4c5hs4509g37l7y9kmb1ysd1916jjrj-merkaartor-0.19.0.drv' failed with exit code 1;
       last 10 log lines:
       > -- Checking for modules 'gdal;proj;exiv2;gpsd'
       > --   No package 'gpsd' found
       > CMake Error at /nix/store/yi235g10sp8jx939zpfli0s74154ph3v-cmake-3.26.4/share/cmake-3.26/Modules/FindPkgConfig.cmake:607 (message):
       >   A required package was not found
       > Call Stack (most recent call first):
       >   /nix/store/yi235g10sp8jx939zpfli0s74154ph3v-cmake-3.26.4/share/cmake-3.26/Modules/FindPkgConfig.cmake:829 (_pkg_check_modules_internal)
       >   CMakeLists.txt:523 (pkg_check_modules)
       >
       >
       > -- Configuring incomplete, errors occurred!
```

**Comment by sikmir** [comment]
What about adding the patch? Since no response from upstream.

File: pkgs/applications/misc/merkaartor/default.nix [hunk 2]
```diff
@@ -32,37 +33,37 @@ stdenv.mkDerivation rec {
       url = "https://github.com/openstreetmap/merkaartor/commit/1e20d2ccd743ea5f8c2358e4ae36fead8b9390fd.patch";
       hash = "sha256-aHjJLKYvqz7V0QwUIg0SbentBe+DaCJusVqy4xRBVWo=";
     })
+    # https://github.com/openstreetmap/merkaartor/pull/290
+    ./0001-Fix-build-with-GPSD.patch
```
Comment (sikmir) [inline pkgs/applications/misc/merkaartor/default.nix:37]:
```suggestion
    # https://github.com/openstreetmap/merkaartor/pull/290
    (fetchpatch {
      url = "https://github.com/openstreetmap/merkaartor/commit/7dede77370d89e8e7586f6ed5af225f9b5bde6cf.patch";
      hash = "...";
    })
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 259434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>merkaartor</li>
  </ul>
</details>

**Review by sikmir — APPROVED** [review]

**Pull request merged**